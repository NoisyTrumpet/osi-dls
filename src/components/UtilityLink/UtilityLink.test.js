import React from "react";
import { render, screen } from "@testing-library/react";
import {
  toBeInTheDocument,
  toHaveClass,
  toHaveAttribute
} from "@testing-library/jest-dom/matchers";
import { UtilityLink } from "./";

expect.extend({
  toBeInTheDocument,
  toHaveClass,
  toHaveAttribute
});

describe("UtilityLink", () => {
  // Tests whether the utility link is rendered
  test("renders Utility Link component", () => {
    const { container } = render(
      <UtilityLink
        to="https://www.hallmark.com/"
        title="utility link"
        icon="undefined"
      />
    );
    expect(screen.getByText("utility link").closest("div")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
  // Tests whether it has the utility-link class
  test("parent container should have a class of utility-link", () => {
    render(
      <UtilityLink
        icon="placeholder-bold"
        to="https://www.hallmark.com/"
        title="util"
      />
    );
    expect(screen.getByText("util").closest("div")).toHaveClass("utility-link");
  });
  // Renders a title within the utility link
  test("tests whether the correct title and icon are rendered in utility link", () => {
    const { container, getByText } = render(
      <UtilityLink
        to="https://www.hallmark.com/"
        title="test"
        icon="placeholder-bold"
      />
    );
    expect(getByText("test")).toBeTruthy();
    expect(container).toMatchInlineSnapshot(`
      <div>
        <div
          class="utility-link"
        >
          <a
            href="https://www.hallmark.com/"
          >
            <span
              aria-hidden="true"
              class="icon icon-placeholder-bold icon-24"
              data-testid="icon"
            />
            <span
              class="typography typography-body typography-body-medium"
            >
              test
            </span>
          </a>
        </div>
      </div>
    `);
  });
});
