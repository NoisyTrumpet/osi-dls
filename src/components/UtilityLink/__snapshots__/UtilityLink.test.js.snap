// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UtilityLink renders Utility Link component 1`] = `
<div
  class="utility-link"
>
  <a
    href="https://www.hallmark.com/"
  >
    <span
      aria-hidden="true"
      class="icon icon-undefined icon-24"
      data-testid="icon"
    />
    <span
      class="typography typography-body typography-body-medium"
    >
      utility link
    </span>
  </a>
</div>
`;
